src/language-js/printer-estree.js

```diff
@@ -3863,40 +3863,33 @@ function isSimpleFlowType(node) {
   );
 }
 
-const functionCompositionFunctionNames = new Set([
-  "pipe", // RxJS, Ramda
-  "pipeP", // Ramda
-  "pipeK", // Ramda
-  "compose", // Ramda, Redux
-  "composeFlipped", // Not from any library, but common in Haskell, so supported
-  "composeP", // Ramda
-  "composeK", // Ramda
-  "flow", // Lodash
-  "flowRight", // Lodash
-  "connect", // Redux
-  "createSelector" // Reselect
-]);
-const ordinaryMethodNames = new Set([
-  "connect" // GObject, MongoDB
-]);
-
-function isFunctionCompositionFunction(node) {
-  switch (node.type) {
-    case "OptionalMemberExpression":
-    case "MemberExpression": {
-      return (
-        isFunctionCompositionFunction(node.property) &&
-        !ordinaryMethodNames.has(node.property.name)
-      );
-    }
-    case "Identifier": {
-      return functionCompositionFunctionNames.has(node.name);
-    }
-    case "StringLiteral":
-    case "Literal": {
-      return functionCompositionFunctionNames.has(node.value);
+function looksLikeFunctionCompositionArgs(args) {
+  if (args.length <= 1) {
+    return false;
+  }
+  let functionCount = 0;
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+    if (
+      arg.type === "FunctionExpression" ||
+      arg.type === "ArrowFunctionExpression"
+    ) {
+      if (++functionCount > 1) {
```

Comment:
It's just a nit, but can you separate the increase from the `if` check? I personally like that, but we try to avoid those in codebase

Reply:
My bad, I had it split up at first, bu noticed y’all had prefix increment operators in a couple places and changed it.